### PR TITLE
Support in Set-GitHubContent to upload binary files

### PR DESCRIPTION
**Pull request by StanleyGoldman**
#### Description
As mentioned in #335, this pull request adds functionality to upload binary content with `Set-GitHubContent`

#### Checklist

- [x] You actually ran the code that you just wrote, especially if you did just "one last quick change".
- [ ] Comment-based help added/updated, including examples.
- [ ] [Static analysis](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#static-analysis) is reporting back clean.
- [ ] New/changed code adheres to our [coding guidelines](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#coding-guidelines).
- [ ] [Formatters were created](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#formatters) for any new types being added.
- [ ] New/changed code continues to [support the pipeline](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#pipeline-support).
- [ ] Changes to the manifest file follow the [manifest guidance](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#module-manifest).
- [ ] Unit tests were added/updated and are all passing. See [testing guidelines](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#testing).  This includes making sure that all pipeline input variations have been covered.
- [ ] Relevant usage examples have been added/updated in [USAGE.md](https://github.com/microsoft/PowerShellForGitHub/blob/master/USAGE.md).
- [ ] If desired, ensure your name is added to our [Contributors list](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#contributors)

### CODE REVIEW

File: GitHubContents.ps1 [hunk 1]
```diff
@@ -358,7 +362,12 @@ filter Set-GitHubContent
 
     $uriFragment = "/repos/$OwnerName/$RepositoryName/contents/$Path"
 
-    $encodedContent = [Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes($Content))
+    if ($ContentPath) {
+        $encodedContent = [Convert]::ToBase64String((Get-Content -path $ContentPath -AsByteStream -Raw))
```
Comment (HowardWolosky) [inline GitHubContents.ps1:366]:
```suggestion
        $encodedContent = [Convert]::ToBase64String((Get-Content -Path $ContentPath -AsByteStream -Raw))
```

File: GitHubContents.ps1 [hunk 2]
```diff
@@ -358,7 +362,12 @@ filter Set-GitHubContent
 
     $uriFragment = "/repos/$OwnerName/$RepositoryName/contents/$Path"
 
-    $encodedContent = [Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes($Content))
+    if ($ContentPath) {
```
Comment (HowardWolosky) [inline GitHubContents.ps1:365]: Style for this project has brackets on an empty line
```suggestion
    if ($ContentPath)
    {
```

File: GitHubContents.ps1 [hunk 3]
```diff
@@ -358,7 +362,12 @@ filter Set-GitHubContent
...
+        $encodedContent = [Convert]::ToBase64String((Get-Content -path $ContentPath -AsByteStream -Raw))
+    }
+    else {
```
Comment (HowardWolosky) [inline GitHubContents.ps1:368]:
```suggestion
    else
    {
```

File: GitHubContents.ps1 [hunk 4]
```diff
@@ -340,6 +342,8 @@ filter Set-GitHubContent
 
         [string] $AuthorEmail,
 
+        [string] $ContentPath,
```
Comment (HowardWolosky) [inline GitHubContents.ps1:345]: Parameters should be ordered in the same position as their documentation.  In this case, this should appear right below `[string] $Content`.

Comment (HowardWolosky) [inline GitHubContents.ps1:345]: `$ContentPath` should be mutually exclusive with `$Content`.  You can introduce an additional `ParameterSet` to help control that.  It's worthy of a separate example in the documentation above as well.

File: GitHubContents.ps1 (hunk 1 above)
Comment (HowardWolosky) [inline GitHubContents.ps1:366]: We should also have a new [test](https://github.com/microsoft/PowerShellForGitHub/blob/master/Tests/GitHubContents.tests.ps1) that validates this new functionality as well.

File: GitHubContents.ps1 [hunk 5]
```diff
@@ -322,7 +325,6 @@ filter Set-GitHubContent
         [string] $CommitMessage,
 
         [Parameter(
-            Mandatory,
```
Comment (HowardWolosky) [inline GitHubContents.ps1:325]: This needs to remain mandatory in its own `ParameterSet`.

File: GitHubContents.ps1 [hunk 6]
```diff
@@ -322,7 +325,6 @@ filter Set-GitHubContent
         [string] $CommitMessage,
 
         [Parameter(
```
Comment (HowardWolosky) [inline GitHubContents.ps1:327]: We may also want to allow this to be `ValueFromPipeline` to enable the content to be piped directly into this method.  That's certainly an optional change for this PR though.

**Review by HowardWolosky — CHANGES_REQUESTED** [review]
Welcome to the project @StanleyGoldman, and thanks for the idea and for the submission.  You're off to a great start.  I've provided some feedback for you to take a look at when you get a chance.

Thanks again.

File: GitHubContents.ps1 (hunk 5 above)
Comment (StanleyGoldman) [reply GitHubContents.ps1:325]: Fixed

File: GitHubContents.ps1 (hunk 4 above)
Comment (StanleyGoldman) [reply GitHubContents.ps1]: Fixed

Comment (StanleyGoldman) [reply GitHubContents.ps1]: Fixed

File: GitHubContents.ps1 (hunk 1 above)
Comment (StanleyGoldman) [reply GitHubContents.ps1]: Added a test

File: GitHubContents.ps1 [hunk 7]
```diff
@@ -231,6 +231,9 @@ filter Set-GitHubContent
     .PARAMETER Content
         The new file content.
 
+    .PARAMETER ContentPath
+        The local path to file content.
```
Comment (HowardWolosky) [inline GitHubContents.ps1:235]:
```suggestion
        The local path to a file whose content should replace what is currently stored remotely in the repo at the Path location.
```

File: GitHubContents.ps1 (hunk 5 above)
Comment (HowardWolosky) [reply GitHubContents.ps1:325]: Take a look at `Get-GitHubEvent`.  That's an example where we had to explode the number of ParameterSets in order to support both the `Elements` and `Uri` repo scenarios while still accurately separating out the different possible inputs for the method itself.

**Review by HowardWolosky — CHANGES_REQUESTED** [review]
Hey @StanleyGoldman,

Thanks so much for these updates, and my sincerest apologies for such a long time to get back to you with a follow-up review.  I completely missed when you pushed up fixes earlier.

The main feedback here is still around the ParameterSets.  Take a look at my more detailed comment in the reactivated comment thread to see if that gives you the additional information that you need to get it fixed.

**Pull request closed without merging**